### PR TITLE
[api] Add new /mkdir public API

### PR DESCRIPTION
**Pull request by Harshg999**
## What changes were proposed in this pull request?

- Implement a dedicated /mkdir API method tailored for new storage browser.
- This separates the old method for existing filebrowser to not have regressions + much cleaner new method.

## How was this patch tested?

- Tested manually

### CODE REVIEW

File: apps/filebrowser/src/filebrowser/api.py [hunk 1]
```diff
@@ -80,3 +83,19 @@ def get_filesystems_with_home_dirs(request): # Using as a public API only for no
...
   return JsonResponse(filesystems, safe=False)
+
+
+@error_handler
+def mkdir(request):
+  path = request.POST.get('path')
+  name = request.POST.get('name')
+
+  if posixpath.sep in name or "#" in name:
+      raise Exception(_("Could not name folder \"%s\": Slashes or hashes are not allowed in filenames." % name))
```
Comment (bjornalm) [inline apps/filebrowser/src/filebrowser/api.py:94]: 1. Are these the only restrictions for folder names across all supported platforms?
2. I think we use the new error message structure for new API end points

**Review by bjornalm — COMMENTED** [review]
Nice work, I agree it is a good idea to clean up the file browser APIs

**Review by nidhibhatg — APPROVED** [review]

File: apps/filebrowser/src/filebrowser/api.py (hunk 1 above)
Comment (Harshg999) [reply apps/filebrowser/src/filebrowser/api.py]: 1. These are the common restrictions on folder names. FS specific restrictions are checked in respective mkdir methods.
2. It still needs the core error map. Let's convert these messy APIs and interatively take up new error message roaditem.

**Review by JohanAhlen — APPROVED** [review]
LGTM!

**Review by bjornalm — APPROVED** [review]

**Pull request merged**